### DISCUSSION TITLE
SMARTS output for radicals?

### DISCUSSION THREAD

**Discussion by jepdavidson** [rdkit/rdkit#4613] [Q&A]
Dear All,

I was recently looking at converting some MDL molblocks with query features into SMARTS.
An issue that I hit was how radicals are handled.  Here's an example:

```python
from rdkit import Chem

mb1 = '''
  -ISIS-  10230216212D

  1  0  0  0  0  0  0  0  0  0999 V2000
   -0.5625   -4.2833    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0
M  RAD  1   1   2
M  END
'''

m1 = Chem.MolFromMolBlock(mb1, removeHs=False)
sma1 = Chem.MolToSmarts(m1)
print(sma1)
```
```
>>> [#7]
```

The RDKit query molecule (m1) recognises the RAD tag absolutely fine, and behaves as expected if being used to find molecules with radical nitrogens.  It is just that this information is lost on converting to SMARTS.

Here's another example, but this time with a MOL that is carrying a positive charge instead of a radical (not important - but this MOL is from Marvin, not ISIS!).
```python
mb2 = '''
  Mrv2019 10152115502D          

  1  0  0  0  0  0            999 V2000
   -5.7143   -3.8393    0.0000 N   0  3  0  0  0  0  0  0  0  0  0  0
M  CHG  1   1   1
M  END
'''

m2 = Chem.MolFromMolBlock(mb2, removeHs=False)
sma2 = Chem.MolToSmarts(m2)
print(sma2)
```
```
>>> [#7+]
```

So what I am wondering is whether the radical tags should find their way into output SMARTS in the same way that charges do?
Something like ```[#7;v2+0]``` in this case(?).

Or is it really by design that this isn't currently done (there is every chance that I have missed something obvious and/or haven't given this enough thought!)?

(I am starting to wonder if there is a problem (philosophical, not technical) explicitly taking uncharged to mean +0 in SMARTS output...)

Kind regards

James

**Comment by greglandrum** [comment]
Oh, that’s an excellent question.
The most explicit way to support this would be via CXSMILES extensions, those allow radical counts to be directly expressed. The downside of that, of course, is that it’s not standard SMARTS.

Using the combo of making the valence and charge explicit seems like a reasonable workaround using plain SMARTS but, as you point out, it may be over interpreting some queries.

I’d be interested in hearing other peoples’ opinions/ideas. @bp-kelley any thoughts?
